order-book/src/test/java/com/market/banica/order/book/service/grpc/componentTests/OrderBookComponentIT.java

```diff
@@ -200,12 +316,135 @@ public void cancelItemSubscription(CancelSubscriptionRequest request, StreamObse
         assertNull(auroraClient.getCancellableStubs().get(productName));
     }
 
+    private void startOrderBookServiceWithGetOrderBookItemLayersMethodOverridden(ItemOrderBookResponse response) {
+        new Thread(() -> {
+            try {
+                Thread.sleep(THREAD_SLEEP_TIME_DEFAULT);
+                grpcCleanup.register(InProcessServerBuilder
+                        .forName(serverName).directExecutor().addService(testConfiguration.getGrpcOrderBookServiceItemLayers(response))
+                        .build().start());
+            } catch (InterruptedException | IOException e) {
+                e.printStackTrace();
+            }
+        }).start();
+    }
+
+    private void startAuroraServiceWithRequestMethodOverridden() {
+        new Thread(() -> {
+            try {
+                Thread.sleep(THREAD_SLEEP_TIME_DEFAULT);
+                grpcCleanup.register(InProcessServerBuilder
+                        .forName(serverNameThree).directExecutor().addService(new AuroraServiceGrpc.AuroraServiceImplBase() {
+                            @Override
+                            public void request(Aurora.AuroraRequest request, StreamObserver<Aurora.AuroraResponse> responseObserver) {
+                                String[] topicSplit = request.getTopic().split(DELIMITER);
+
+                                ItemOrderBookRequest build = populateItemOrderBookRequest(request.getClientId(), topicSplit[1], Long.parseLong(topicSplit[2]));
+
+                                ItemOrderBookResponse orderBookItemLayers = blockingStub.getOrderBookItemLayers(build);
+
+                                responseObserver.onNext(Aurora.AuroraResponse
+                                        .newBuilder()
+                                        .setMessage(Any.pack(orderBookItemLayers))
+                                        .build());
+                                responseObserver.onCompleted();
+                            }
+                        })
+                        .build().start());
+            } catch (InterruptedException | IOException e) {
```

Comment:
in tests you can just add the exception to be thrown by the method/test, it is not required to be catched and print the stacktrace (it is in a few places in the class, will not write a comment on each place)

Reply:
exceptions are local to a thread and main thread doesn't actually see the run method